aioftp/client.py

```diff
@@ -338,6 +341,188 @@ def parse_directory_response(self, s):
 
         return pathlib.PurePosixPath(directory)
 
+    @staticmethod
+    def parse_unix_mode(s):
+        """
+        Parsing unix mode strings("rwxr-x--t") into
+        hexacimal notation
+
+        :param s: mode string
+        :type s: :py:class:`str`
+
+        :return mode:
+        :rtype: :py:class:`int`
+        """
+        def parse_rw(s):
+            mode = 0
+
+            if s[0] == 'r':
+
+                mode |= 4
+
+            elif s[0] != '-':
+
+                raise ValueError
+
+            if s[1] == 'w':
+
+                mode |= 2
+
+            elif s[1] != '-':
+
+                raise ValueError
+
+            return mode
+
+        mode = 0
+        mode |= parse_rw(s[0:2]) << 6
+        mode |= parse_rw(s[3:5]) << 3
+        mode |= parse_rw(s[6:8])
+
+        if s[2] == 's':
+
+            mode |= 0o4100
+
+        elif s[2] == 'x':
+
+            mode |= 0o0100
+
+        elif s[2] != '-':
+
+            raise ValueError
+
+        if s[5] == 's':
+
+            mode |= 0o2010
+
+        elif s[5] == 'x':
+
+            mode |= 0o0010
+
+        elif s[5] != '-':
+
+            raise ValueError
+
+        if s[8] == 't':
+
+            mode |= 0o1000
+
+        elif s[8] == 'x':
+
+            mode |= 0o0001
+
+        elif s[8] != '-':
+
+            raise ValueError
+
+        return mode
+
+    @staticmethod
+    def parse_ls_date(s):
+        """
+        Parsing dates from the ls unix utility
+        ("Nov 18  1958" and "Nov 18 12:29")
+
+        :param s: ls date
+        :type s: :py:class:`str`
+
+        :rtype: :py:class:`str`
+        """
+        if not (s[3] == ' ' and s[6] == ' '):
+
+            raise ValueError
+
+        month = "{0:02d}".format(time.strptime(s[:3], "%b").tm_mon)
```

Comment:
the problem here is that this is locale-dependent, i.e. if the locale on server is (for example) ru_RU and on the client en_US - it will just fail. I don't see a good fix for this though.